Added code to enable/disable motion detection for foscam cameras

viswa-swami opened the pull request:
## Description:
This support to enable/disable motion detection was added in 0.48.1 as a generic service for cameras. Motion detection can be enabled/disabled for camera component.

This code-set will facilitate users to enable motion detection or disable motion detection Foscam cameras. 

With the motion detection disabled, users can just stream the video data without being notified about motion detection.   When the motion_detection is enabled, then users will be notified when motion is detected by the camera.

Tested with the foscam cameras in my home.

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#<home-assistant.github.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml

```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

viswa-swami committed:
Added support to enable/disable motion detection for foscam cameras. This support was added in 0.48.1 as a generic service for cameras. Motion detection can be enabled/disabled for foscam cameras with this code-set.

viswa-swami committed:
Fixed the violation identified by hound-bot

balloob committed:
Update frontend

balloob committed:
Fix iFrame panel test

viswa-swami committed:
Fixed the comment posted by HoundCI-Bot regarding using imperative mood statement for pydocstyle

viswa-swami committed:
Fixed the error that travis-ci bot found.

MartinHjelmare committed:
Make deps directory persistent over upgrades (#7801)

* Use pip install --user if venv not active

* Set PYTHONUSERBASE to deps directory, when installing with --user
  option.
* Reset --prefix option to workaround incompatability when installing
  with --user option. This requires pip version 8.0.0 or greater.
* Require pip version 8.0.3.
* Do not delete deps directory on home assistant upgrade.
* Fix local lib mount and check package exist.

* Update and add tests

* Fix upgrade from before version 0.46

* Extract function to get user site

* Add function(s) to package util to get user site.
* Use async subprocess for one of the functions to get user site.
* Add function to package util to check if virtual environment is
  active.
* Add and update tests.

* Update version for last removal of deps dir

* Address comments

* Rewrite package util tests with pytest

* Rewrite all existing unittest class based tests for package util as
  test functions, and capitalize pytest fixtures.
* Add test for installing with target inside venv.

amelchio committed:
LIFX: support for multizone (#8399)

* Make aiolifx modules easily available

* Use aiolifx features_map for deciding bulb features

Also move the feature detection out of Light so it is available even
during the initial detection.

* Move each LIFX light type to a separate class

* Simplify AwaitAioLIFX

This has become possible with recent aiolifx that calls the callback even
when a message is lost.

Now the wrapper can be used also before a Light is added though the register
callback then has to become a coroutine.

* Refactor send_color

* Add support for multizone

This lets lifx_set_state work on individual zones.

Also update to aiolifx_effects 0.1.1 that restores the state for individual
zones.

d0ugal committed:
Add RGB support to switch.flux (#8417)

dersger committed:
Fix media_position for cast component (#8452)

* Make it available during state paused.

* Don't adjust for media_position_updated_at. I.e. do as vlc, sonos etc
  so that returned position is the position at the time of
  media_position_updated_at, not now.

gollo committed:
Remove km from visibility, add visibility_distance (#8454)

* Remove km from visibility, add visibility_distance

* Fix line length

* Fix trailing space and line break indentation

* Indentation

* More whitespace

hoopty committed:
Radarr sensor fix for issue #8250 (#8456)

* Radarr sensor fix for issue #8250

* Radarr sensor fix for issue #8250

balloob commented on homeassistant/components/camera/foscam.py:
```diff
@@ -78,3 +82,63 @@ def camera_image(self):
+        # trigger interval: interval between each motion triggers
+        # schedule: set schedule for days. default is 24x7
+        # area: sort of like zones. default is all areas visible to cam
+        uri_template = self._base_url \
```
This is too much protocol specific information inside Home Assistant. Please use an existing 3rd party lib to control Foscam or create a new one.

amelchio committed:
Accept transition for light.toggle (#8466)

andrey-git committed:
Make themes API work even when themes are not defined. (#8473)

MartinHjelmare committed:
Fix check for running inside venv (#8481)

* Import and use the function from pip instead of defining it
  ourselves.
* Fix tests.

balloob committed:
Version bump to 0.50.0dev0

Commit:
Bump to KNXIP 0.5 (#8492)

* Bump to KNXIP 0.5

* Updated requirements file

balloob committed:
Expose all components on hass [Concept] (#8490)

* Add components concept

* Lint

* Raise ImportError if component not found

balloob committed:
Add bind_hass to components (#8502)

* Add bind_hass to components

* Add bind_hass to group

maikelwever committed:
Fix TP-Link device tracker regression since 0.49 (#8497)

* Fix TP-Link device tracker regression since 0.49

This regression was introduced by #8322.

Fix is to utf encode the password like the other TP-Link backends do.

* Fix linting issue introduced in previous commit

Commit in question: 677f3fbb7f821ee925364c8260d235dce4f0ddbe

balloob committed:
Persistent notification import (#8507)

* Rewrite persistent notification creation

* Update components.is_on to use auto loading

* Fix two hass parameters

Commit:
Added additional attributes to OwnTracks device_tracker (#8503)

* Added additional attributes to OwnTracks device_tracker

* Added missing space after :

pvizeli committed:
Update dlib_face_detect.py (#8516)

rcloran committed:
prometheus: Convert fahrenheit to celsius (#8511)

pvizeli committed:
Realfix for dlib (#8517)

* Update dlib_face_detect.py

* fix lint

* Update dlib_face_detect.py

azogue committed:
Attach the `chat_id` for a callback query from a chat group (fixes #8461) (#8523)

fronzbot committed:
Added sensor state rounding (#8499)

MikeChristianson committed:
fix for a bug introduced with media support in #8282 (#8513)

data may be None if twitter data property unconfigured:
  File "/opt/homeassistant/homeassistant_venv/lib/python3.4/site-packages/homeassistant/components/notify/twitter.py", line 63, in send_message
    media = data.get(ATTR_MEDIA)

fabaff committed:
Fix KeyError

azogue committed:
handle timeout errors without logging.exception when updating hue lights; double quotes in log msgs (#8524)

amelchio committed:
Remove deprecated automation keywords (#8510)

* Remove deprecated automation keywords

* Remove retired test case

* Remove retired keyword

janLo committed:
Citybikes: Allow None as result for empty slots (#8528)

The citibykes API returns "null" as value for empty_slots on some
stations (see #8527). This causes the component to not process the data.

This is fixed by accepting None as valid data. The row in the frontend
is left empty if "null" was returned by the service.

fixes #8527

phil-lavin committed:
Return a 0 temperature value when none is found (#8518)

* Return a 0 temperature value when none is found

It's well documented that these TRVs will only return the current temperature
for a short time after the actuator has moved. This means that, usually, they will
not return the current temperature. Setting a non-value here causes errors in the logs
and for the TRV to not show on the dashboard at all

* Fix lint issue

titilambert committed:
Fix #6469 and #6828 (#8537)

fabaff committed:
Update docstrings (#8536)

fabaff committed:
Upgrade TwitterAPI to 2.4.6 (#8535)

postlund committed:
Fix support for multiple Apple TVs (#8539)

titilambert committed:
Decora: Fix set brightness and improve reconnection (#8522)

balloob committed:
Update demo.py

fronzbot committed:
Initial support for Google Wifi/OnHub (#8485)

* Initial support for Google Wifi/OnHub

* Moved state logic to update function of API class

- Throttle added to update
- State logic implementation is cleaner
- Modified tests to work with the new throttle on update

tchellomello committed:
Fix exception dlib_face_identify when image is not recognized by face_recognition module (#8552)

*   Some images are not supported by face_recognition, so this patch treats the error
  messages instead throwing a traceback. Fixes #7867

* Makes lint happy

minchik committed:
[media_extractor] Add support for custom stream queries for media_extractor (#8538)

* Add support for different stream formats

* Encapsulate logic inside MediaExtractor class

* Add CONFIG_SCHEMA

* Fix for cases when youtube-dl returns content of playlist as list

jawilson committed:
Report Harmony remote off if state is unknown (#8547)

Add an optional extended description…

viswa-swami committed:
Upgrade the alarmdecoder dependency library from 0.12.1 to 0.12.3. (#8542)

* Upgrade the alarmdecoder dependency library from 0.12.1 to 0.12.3. Nutech software who owns this library have upgraded this library with some fixes regarding arming it to home/away and then disarming the alarm. Without this upgraded library, HASS is having a problem when we try to disarm an armed alarm after around 8 hours or so.

* Updated the requirements_all.txt by running the script gen_requirements_all.py

azogue committed:
turn_on_action and turn_off_action with script syntax (#8558)

tchellomello committed:
Extends Pi-hole sensor to support the new sensors: (#8549)

- domains_being_blocked
  - queries_cached
  - queries_forwarded
  - unique_clients
  - unique_domains

Commit:
Support for Wink Switch and Light groups also fix fan speed selection (#8501)

* Support for Switch and Light groups, fix fan speed

* Fixed hound violations

Commit:
Added a service to write to KNX group addressed including documentation (#8491)

* Added a service to write to KNX group addressed including documentation

* Define parameters as required

* Reformating

* Moved service documentation to service.yaml

* Moved service documentation to services.yaml

* Update knx.py

jawilson committed:
Handle manual edits to emulated_hue_ids.json (#8560)

amelchio committed:
LIFX: assume default features for unknown products (#8553)

This makes the detection work for prototypes as well.

ypollart committed:
Fix broken status update for lighting4 devices (#8543)

* Fix broken status update for lighting4 devices

* Fixed indentation

sdague committed:
Enhance python_script to support "_getitem_" (#8541)

* Enhance python_script to support "_getitem_"

In order to use dict / list structures in python scripts we need
_getitem_ allowed in the RestrictedPython environment. There is a
default_guarded_getitem included with RestrictedPython, which is a
pass through used in the Eval code paths.

* Add tests for dict/list support in python_scripts

* Lint

balloob committed:
Update persistent deps dir version in config.py (#8479)

* Update persistent deps dir version in config.py

* Update last version to remove deps dir in tests

Danielhiversen committed:
Add is_lighting4 to RfxtrxBinarySensor (#8563)

pavoni committed:
Bump pyver to fix exception in fetching colours.

pavoni committed:
Merge pull request #8564 from home-assistant/bump_pyvera

Bump pyvera to fix colour exception

Danielhiversen committed:
Xiaomi gw support (#8555)

* xiaomi support

* xiaomi support

* style

* style

* style

* style

* style

* coveragerc

* Update xiaomi.py

* Update xiaomi.py

* Update xiaomi.py

* refactorization

* refactorization

* config validation

* style

* package

* refactorization

* refactorization

* refactorization

* HA integration

namadori committed:
fix #8263 corrected Adafruit DHT library version from 1.3.0 to 1.3.2 (#8562)

Add an optional extended description…

viswa-swami committed:
Merge branch 'enable_disable_motion_detection_foscam_cameras' of https://github.com/viswa-swami/home-assistant into dev

viswa-swami committed:
As per comment from @balloob, Instead of directly using the URL to talk to foscam, used a 3rd party foscam library to communicate with it. This library already has support to enable/disable motion detection and also APIs to change the motion detection schedule etc. Need to add more support in the pyfoscam 3rd party library for checking if motion was detected or even if sound was detected. Once that is done, we can add that into HASS as well.

viswa-swami commented:
Oops... i think my branch went bad !!!!!! Apologies for any confusion.... I can possibly close this PR and create a fresh branch and commit back the changes for fresh PR.
